### PR TITLE
test: basic tests for generators

### LINKED ISSUE #65: Tests for generators
Implement tests for generators:

- Application (vue3-vite)
- Component
- Library
- Vitepress Docs
- Cypress

### PR DESCRIPTION
**Pull request by abemscac**
Close #65

### CODE REVIEW

File: packages/vue3-vite/src/generators/docs/files/README.md.template [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# <% projectTitle %> Documentation
+# <%= projectTitle %> Documentation
```
Comment (abemscac) [inline packages/vue3-vite/src/generators/docs/files/README.md.template:1]: I accidentally found that some equal signs are missing in the template files of docs.

Comment (sampullman) [reply packages/vue3-vite/src/generators/docs/files/README.md.template:1]: Good catch!

File: packages/vue3-vite/src/generators/library/files/src/index.ts.template [hunk 2]
```diff
@@ -1,3 +1,3 @@
 export {
-  Widget
-} from '@lib/Widget.vue'
\ No newline at end of file
+  MyWidget
+} from '@lib/MyWidget.vue'
```
Comment (abemscac) [inline packages/vue3-vite/src/generators/library/files/src/index.ts.template:3]: I have to give it a multi-word name to make the lint test pass.
`Cookies` and `Home` are also renamed to `CookiesConsent` and `HomePage` respectively for the same reason.

**Comment by abemscac** [comment]
Hi @sampullman 

I'm not sure if I'm doing it correctly; I wonder what is the difference between unit-test and e2e-test in a repository like this one?
I added some flag test case (i.e. `--title` for cypress generator) in `packages/vue3-vite/src/generators/cypress/generator.spec.ts`; is that the right place to do such thing? :thinking:

**Comment by sampullman** [comment]
Oh! Sorry for the confusion, if the test is running CLI commands it would be considered E2E.

In the unit tests we should only call generator/executor functions directly. This is how I think about it:
- Unit tests call functions in a single module and verify the result. Everything except the module being tested should be mocked (where possible)
- E2E tests try to run the project how users would, i.e. via the command line

**Comment by abemscac** [comment]
I see.
I'll try to adjust the tests to better align with the design/definition, thank you!

**Comment by sampullman** [comment]
Sounds good - I can also try to write a better description of testing in the README before merging as well.

**Comment by abemscac** [comment]
Hi @sampullman 

I think this PR is ready for review, thank you!

**Comment by sampullman** [comment]
Nice work! I added the e2e tests to Github actions, but it seems like there are some issues running in CI. I think I'll merge now and create an issue for fixing them.

I updated packages as well, so we'll need to migrate to Vite 4 and Cypress 12!

File: e2e/vue3-vite-e2e/jest.config.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
   displayName: 'vue3-vite-e2e',
   preset: '../../jest.preset.ts',
-  globals: {
-    'ts-jest': {
-      tsconfig: '<rootDir>/tsconfig.spec.json',
-    },
-  },
   transform: {
-    '^.+\\.[tj]s$': 'ts-jest',
+    '^.+\\.[tj]s?$': [
```
Comment (sampullman) [inline e2e/vue3-vite-e2e/jest.config.ts:5]: Update tsconfig to new transform format (for `ts-jest`).

File: e2e/vue3-vite-e2e/tests/cypress.spec.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import {
+  checkFilesExist,
+  runNxCommandAsync,
+  uniq,
+} from '@nrwl/nx-plugin/testing';
+
+jest.setTimeout(60000);
```
Comment (sampullman) [inline e2e/vue3-vite-e2e/tests/cypress.spec.ts:7]: Increase timeout to 60s to see if it helps CI pass.

File: e2e/vue3-vite-e2e/tests/docs.spec.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import {
+  checkFilesExist,
+  readFile,
+  runNxCommandAsync,
+  uniq,
+} from '@nrwl/nx-plugin/testing';
+
+jest.setTimeout(60000);
+
+describe('docs e2e', () => {
```
Comment (sampullman) [inline e2e/vue3-vite-e2e/tests/docs.spec.ts:10]: Skipping this test for now, the doc generator will need to be updated at some point. I think the current version breaks under Vite 4.

**Pull request merged**